Mon Mar 26 23:48:57 FET 2012  Sergei Trofimovich <slyfox@community.haskell.org>
  * compiler/GF/Text/Coding.hs: fix build failure against ghc-7.2
diff -rN -u old-GF/src/compiler/GF/Text/Coding.hs new-GF/src/compiler/GF/Text/Coding.hs
--- old-GF/src/compiler/GF/Text/Coding.hs	2012-03-27 20:10:08.351852807 +0300
+++ new-GF/src/compiler/GF/Text/Coding.hs	2012-03-27 20:10:19.437714215 +0300
@@ -1,3 +1,5 @@
+{-# LANGUAGE CPP #-}
+
 module GF.Text.Coding where
 
 import qualified Data.ByteString as BS
@@ -23,7 +25,11 @@
   where
     translate cod cbuf
       | i < w     = do bbuf <- newByteBuffer 128 WriteBuffer
+#if __GLASGOW_HASKELL__ >= 702
+                       (_,cbuf,bbuf) <- cod cbuf bbuf
+#else
                        (cbuf,bbuf) <- cod cbuf bbuf
+#endif
                        if isEmptyBuffer bbuf
                          then ioe_invalidCharacter
                          else do let bs = PS (bufRaw bbuf) (bufL bbuf) (bufR bbuf-bufL bbuf)
@@ -46,7 +52,12 @@
                                             return s
   where
     translate cod bbuf cbuf
-      | i < w     = do (bbuf,cbuf) <- cod bbuf cbuf
+      | i < w     = do
+#if __GLASGOW_HASKELL__ >= 702
+                       (_,bbuf,cbuf) <- cod bbuf cbuf
+#else
+                       (bbuf,cbuf) <- cod bbuf cbuf
+#endif
                        if isEmptyBuffer cbuf
                          then ioe_invalidCharacter
                          else unpack cod bbuf cbuf
